modules/javafx.graphics/src/main/java/javafx/scene/layout/Region.java

```diff
@@ -393,15 +395,13 @@ private static double snapPosition(double value, boolean snapToPixel, double sna
      * @return value either as passed, or floored or ceiled with scale, based on snapToPixel
      */
     private double snapPortionX(double value, boolean snapToPixel) {
-        if (!snapToPixel || value == 0) return value;
-        double s = getSnapScaleX();
-        value *= s;
-        if (value > 0) {
-            value = Math.max(1, Math.floor(value + EPSILON));
-        } else {
-            value = Math.min(-1, Math.ceil(value - EPSILON));
+        if (!snapToPixel || value == 0) {
```

Comment:
are snapPortionX/Y methods still used?
a quick check indicates that they are only used by some test via shims.  if that's the case, could these two methods be deleted?

Reply:
Yeah, they were (luckily) only used by HBox/VBox and won't be needed anymore after this change. I didn't fully clean up this PR yet (and I also need to apply the changes to VBox).  It looks like it's becoming more serious though considering the number of problems that were uncovered, so I can finish this up a bit further.

Reply:
Thank you so much for doing this work, @hjohn !